contracts/Company/Extras/GranularTransferController.sol

```diff
@@ -0,0 +1,187 @@
+        address from,
+        address to,
+        uint256 amount
+    );
+
+    event LogForcedTransferExecuted(
+        address token,
+        address from,
+        address to,
+        uint256 amount
+    );
+
+    event LogTokenHolderAccountFrozen(
+        address owner
+    );
+
+    event LogTokenHolderAccountUnfrozen(
+        address owner
+    );
+
+    ////////////////////////
+    // Constructor
+    ////////////////////////
+
+    constructor(
+        Universe universe,
+        address companyLegalRep
+    )
+        public
+        PlaceholderEquityTokenController(universe, companyLegalRep)
+    {}
+
+    //
+    // Implements ITokenController
+    //
+
+    function onTransfer(address broker, address from, address to, uint256 amount)
+        public
+        constant
+        returns (bool allow)
+    {
+        // ask base controller if transfers are enables
+        allow = PlaceholderEquityTokenController.onTransfer(broker, from, to, amount);
+        GovState s = state();
+        // allow forced transfer only by the token controller itself
+        if ((s == GovState.Funded || s == GovState.Closing) && broker == address(this)) {
+            ForcedTransfer storage t = _forcedTransfers[from];
+            // check if forced transfer matches the actual transfer
+            if (t.amount > 0) {
+                // we allow because we checked the following
+                //  (1) the broker is the company
+                //  (2) there's forced transfer defined for from address
+                //  (3) there's a match for to and amount
+                allow = t.amount == amount && t.to == to;
+            }
+            if (allow) {
+                // we return immediately as we'll not check frozen accounts in that case
+                // often it may happend that from address is already frozen
+                return allow;
+            }
+        }
+        // prevent transfer if account is frozen
+        if (allow && s == GovState.Funded) {
+            allow = !(_frozenAddresses[from] || _frozenAddresses[to] || _frozenAddresses[broker]);
+        }
+    }
+
+    function onAllowance(address owner, address spender)
+        public
+        constant
+        returns (uint256)
+    {
+        uint256 overrideAmount = PlaceholderEquityTokenController.onAllowance(owner, spender);
+        // if no override was set by base class check frozen transfer override
+        // the spender must be token controller - this contract
+        if (overrideAmount == 0 && spender == address(this)) {
+            // require(_forcedTransfers[owner].amount > 0, "ON_ALLOWANCE3");
+            // return amount that can be force-transferred
+            return _forcedTransfers[owner].amount;
+        }
+    }
+
+    //
+    // Overrides IContract
+    //
+
+    function contractId() public pure returns (bytes32 id, uint256 version) {
+        return (0xf7e00d1a4168be33cbf27d32a37a5bc694b3a839684a8c2bef236e3594345d70, 0xFF);
```

Comment:
what is this address? I think a comment would be helpful and make it maintanable